### PR TITLE
[PAGOPA-848] Cosmos DB

### PR DESCRIPTION
**Pull request by jacopocarlini**
#### List of Changes
Changing the interactions with the db, switching from azure table storage to cosmos table

#### Motivation and Context
This was required due to the low performances of the azure table storage. The cosmos table enables us to increase its throughput.

#### How Has This Been Tested?
This has been tested manually, through unit tests and integration tests.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: helm/values-dev.yaml [hunk 1]
```diff
@@ -70,6 +70,7 @@ microservice-chart:
     CACHE_SIZE: "1000"
     CACHE_EXPIRATION_TIME: "1s"
     RECEIPTS_TABLE: "pagopadweugpspaymentssareceiptstable"
+    AZURE_TABLES_TABLENAME: "pagopapagopa-d-weu-gpsweugpdpaymentsreceiptstable"
```
Comment (FedericoRuzzier) [inline helm/values-dev.yaml:73]: Remove receipt table

File: performance-test/src/send_rt_workflow.js [hunk 2]
```diff
@@ -124,5 +124,9 @@ export default function (data) {
 
   const response = http.post(url, payload, soapParams);
 
+  if(parseHTML(response.body).find("outcome").get(0).textContent() !== "OK"){
+    console.log(response.body);
+  }
+
```
Comment (FedericoRuzzier) [inline performance-test/src/send_rt_workflow.js:130]: Delete if clause

File: src/main/java/it/gov/pagopa/payments/exception/AppError.java [hunk 3]
```diff
@@ -26,6 +26,22 @@ public enum AppError {
...
+
+  RECEIPT_CONFLICT(
+          HttpStatus.CONFLICT,
+          "Receipt conflict",
+          "The receipt inserted already exists"),
+
+  NOT_ENOUGH_PAGES(
+          HttpStatus.INTERNAL_SERVER_ERROR,
+          "Not enough pages",
+          "There are not enough receipts for the inserted limit and page number"),
```
Comment (FedericoRuzzier) [inline src/main/java/it/gov/pagopa/payments/exception/AppError.java:43]: Delete

File: src/main/java/it/gov/pagopa/payments/service/PartnerService.java [hunk 4]
```diff
@@ -792,7 +758,8 @@ private PaymentOptionModelResponse managePaSendRtRequest(PaSendRTV2Request reque
         this.getReceiptEntity(
             request.getIdPA(),
             request.getReceipt().getCreditorReferenceId(),
-            request.getReceipt().getDebtor());
+            request.getReceipt().getDebtor(),
+            request.getReceipt().getPaymentDateTime().toString());
```
Comment (FedericoRuzzier) [inline src/main/java/it/gov/pagopa/payments/service/PartnerService.java:762]: insert null annotation

File: src/main/java/it/gov/pagopa/payments/service/PaymentsService.java [hunk 5]
```diff
@@ -1,290 +1,167 @@
...
 
-      if (!result.iterator().hasNext()) {
-        throw new AppException(AppError.RECEIPT_NOT_FOUND, organizationFiscalCode, iuv);
-      }
+    @Autowired private TableClient tableClient;
+    @Autowired
+    private GpdClient gpdClient;
 
-      ReceiptEntity receipt = result.iterator().next();
+    public PaymentsService() {}
```
Comment (FedericoRuzzier) [inline src/main/java/it/gov/pagopa/payments/service/PaymentsService.java:37]: check if used

**Review by FedericoRuzzier — COMMENTED** [review]
Fix

**Comment by jacopocarlini** [comment]
update_code

**Review by andrea-deri — APPROVED** [review]
Good job! 🥇

**Pull request merged**